plugins/inventory/docker_containers.py

```diff
@@ -44,13 +44,23 @@
               R(docker connection plugin,ansible_collections.community.docker.docker_connection),
               and C(docker-api) selects the
               R(docker_api connection plugin,ansible_collections.community.docker.docker_api_connection).
+            - When C(docker-api) is used, all Docker daemon configuration values are passed from the inventory plugin
+              to the connection plugin. This can be controlled with I(configure_docker_daemon).
         type: str
         default: docker-api
         choices:
             - ssh
             - docker-cli
             - docker-api
 
+    configure_docker_daemon:
+        description:
+            - Whether to pass all Docker daemon configuration from the inventory plugin to the connection plugin.
+            - Only used when I(connection_type=docker-api).
+        type: bool
+        default: true
```

Comment:
I definitely agree the default should be `true` but it is a change in behavior isn't it? Would that consider it breaking?
Or is the current situation such that it couldn't ever be used effectively anyway (in which case there's no need to wait for major version)?

Reply:
I've thought a bit about it, and I'm still a bit torn; the original issue is a bug report, and I tend to also see the old behavior as a bug. I decided to also add this option to make it possible to revert to the old behavior, so in case someone really needs the old behavior they don't have to stick to an older collection version.

Reply:
fwiw I agree with you; this is the logical/desired behavior, the previous behavior is a bug. It's always hard to tell when people have relied on buggy behavior. I'm not knowledgeable enough about this plugin to know how likely it is that the new default would have drastically (negative) different behavior; as an end user I would probably prefer something like this released sooner.